block/meta-iosched.c

```diff
@@ -288,6 +320,10 @@ void update_stats(struct request_queue *q) {
 	whole_stat++;
 	spin_unlock(&stat_lock);
 	q->stats[group]++;
+
+	//spin_lock(q->queue_lock); //spin_lock_irq(q->queue_lock);
```

Comment:
krinkinmu, thanks a lot! You are right. I've fixed the problem.

jcmvbkbc, yes, atomics suit better here. Thank you for good advice. I also know some more places in my code where I should replace spinlocks with atomics. I'm going to do it later. Thank you!